pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -522,14 +522,24 @@ func (r *ReconcilePerconaServerMongoDB) deleteCfgIfNeeded(cr *api.PerconaServerM
 		return nil
 	}
 
-	sfsName := cr.Name + "-" + "cfg"
+	sfsName := cr.Name + "-" + api.ConfigReplSetName
 	sfs := psmdb.NewStatefulSet(sfsName, cr.Namespace)
 
 	err := r.client.Delete(context.TODO(), sfs)
 	if err != nil && !k8serrors.IsNotFound(err) {
 		return errors.Wrapf(err, "failed to delete sfs: %s", sfs.Name)
 	}
 
+	svc := corev1.Service{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: cr.Name + "-" + api.ConfigReplSetName, Namespace: cr.Namespace}, &svc)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrap(err, "failed to get config service")
+	}
+	err = r.client.Delete(context.TODO(), &svc)
+	if err != nil {
```

Comment:
should we also check for k8serrors.IsNotFound?
I think if object is not exists delete will also return this error

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/465/commits/06699e728959d2b0fc1122951e27090b6bc61286